pkg/steps/source.go

```diff
@@ -683,3 +683,29 @@ func getSourceSecretFromName(secretName string) *coreapi.LocalObjectReference {
 	}
 	return &coreapi.LocalObjectReference{Name: secretName}
 }
+
+func getReasonsForUnreadyContainers(p *coreapi.Pod) string {
+	builder := &strings.Builder{}
+	for _, c := range p.Status.ContainerStatuses {
+		if c.Ready {
+			continue
+		}
+		var reason, message string
+		switch {
```

Comment:
nit: to prevent multiple lookups, lets just `switch state := c.State; state {`

Reply:
sorry, I don't understand. What do we gain from copying the variable?